slides.typ

```diff
@@ -111,54 +112,32 @@
     }
 
     set text(
-        size: 25pt,
+        size: theme.text-size,
+    )
+    let this-slide = current-slide(
+        section: section,
+        logical-slide : logical-slide,
+        subslide : subslide,
     )
-
-    let decoration(position, body) = {
-        let border = 1mm + color
-        let strokes = (
-            header: ( bottom: border ),
-            footer: ( top: border )
-        )
-        block(
-            stroke: strokes.at(position),
-            width: 100%,
-            inset: .3em,
-            text(.5em, body)
-        )
-    }
 
     set page(
         paper: "presentation-16-9",
-        margin: ( x: 1em, y: 4em ),
-        header: locate( loc => {
-            if counter(page).at(loc).first() > 1 {
-                decoration("header", section.at(loc))
-            }
-        } ),
-        header-ascent: 3em,
-        footer: locate( loc => {
-            if counter(page).at(loc).first() > 1 {
-                decoration("footer")[
-                    #short-author #h(10fr)
-                    #short-title #h(1fr)
-                    #date #h(10fr)
-                    #logical-slide.display()
-                ]
-            }
-        } ),
-        footer-descent: 3em,
+        margin: theme.margin,
+        header: theme.header.with(data, this-slide)(),
+        header-ascent: theme.header-ascent,
+        footer: theme.footer.with(data, this-slide)(),
+        footer-descent: theme.footer-descent,
     )
 
     [
         #align(center + horizon,
-            block(fill: color, inset: 1em, radius: 1em, breakable: false,
+            block(fill: theme.color, inset: theme.inset, radius: theme.radius, breakable: false,
                 [
-                    #text(2em)[*#title*] \
-                    #v(1em)
-                    #text(1.5em)[#author] \
-                    #v(1em)
-                    #date
+                    #text(theme.title-text-size)[*#data.title*] \
+                    #v(theme.title-text-space)
+                    #text(theme.author-text-space)[#data.author] \
+                    #v(theme.title-text-space)
+                    #data.date
                 ]
             )
         )
```

Comment:
Worth discussing this further, but I just wanted to comment that we should probably let themes entirely customize the title slide as well (return a content or similar)

Should also be able to use lambdas for this (with the "big state dict" we discussed previously)